src/dmd/lexer.d

```diff
@@ -373,11 +373,19 @@ class Lexer
             case '\r':
                 p++;
                 if (*p != '\n') // if CR stands by itself
+                {
                     endOfLine();
+                    goto skipFourSpaces;
+                }
                 continue; // skip white space
             case '\n':
                 p++;
                 endOfLine();
+                skipFourSpaces:
+                while (*(cast(uint*)p) == 0x20202020) //' ' == 0x20
```

Comment:
Shouldn't this depend on the architecture?
i.e. on 64-bit you could compare with 8 byte (or here whitespace) at a time?

Reply:
Most editors when you press tab write 4 spaces unless you change the settings. If I checked 8 bytes at a time I would miss optimization for 1 tab(common) and reduced optimization for odd number of tabs. While its possible to make use of bigger registers on 64 bit machines I dont believe increased code complexity is worth it.

Reply:
`p` is a byte-aligned pointer right?
Is it defined behavior to dereference a `uint*` that is not 4-byte aligned?

Reply:
Ah yeah that's another potential issue, at least for non-x86 archs.

Reply:
From GNU compiler
"Enables (or disables) reading and writing of 16- and 32- bit values from addresses that are not 16- or 32- bit aligned. By default unaligned access is disabled for all pre-ARMv6, all ARMv6-M and for ARMv8-M Baseline architectures, and enabled for all other architectures"

ARMv7 and ARMv8 processors that are in phones and tablets should work.
https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html